### DISCUSSION TITLE
progress bar for buffered DurationRanges

### DISCUSSION THREAD

**Discussion by wang-bin** [wang-bin/fvp#122] [General]
the default progress bar [only use the max DurationRange.end](https://github.com/flutter/packages/blob/video_player-v2.9.1/packages/video_player/video_player/lib/video_player.dart#L1082-L1094), is it possible to show all ranges in progress bar? this is mpv's progress bar with 2 ranges: 
<img width="253" alt="截屏2024-08-01 13 56 05" src="https://github.com/user-attachments/assets/6d299656-a4b3-495f-acf1-f633bbc839e7">

@abdelaziz-mahdy

**Comment by abdelaziz-mahdy** [comment]
From my knowledge I don't think there is a stream to be able to push data into due to the limitation of video player interface 🤔

Also what's the difference between the two?

**Comment by wang-bin** [comment]
mpv progress bar has 2 thin lines indicating 2 buffered durations

**Reply by abdelaziz-mahdy** [reply]
I understand that, but each buffered line mean, 

One for network cache, and one for cached frames?

**Reply by wang-bin** [reply]
each line is a time range, for example in the screenshot the 1st line is `0s~2s`, the 2nd is `4s~10s`, current position is 8s, cache line color is different from background, so the 2nd line has 2 colors. seeking in buffered lines will reuse the downloaded data

**Reply by abdelaziz-mahdy** [reply]
Okay, so to do it we will need another stream to handle its ranges, but the video_player interface doesn't have that ability, so I don't know how it can be done sadly

**Reply by wang-bin** [reply]
```dart
  _controller.addListener(() {
      final buffered = _controller.value.buffered;
      ...
    });
```

**Comment by mwesigwadi** [comment]
Can this be compatible with Tizen for Samsung.

**Reply by wang-bin** [reply]
no, only debian compatible systems are supported

**Reply by mwesigwadi** [reply]
Is there a way around this? Or anything I can do to add its support?

**Reply by wang-bin** [reply]
You can try https://github.com/flutter-tizen/plugins/tree/master/packages if flutter is not officially supported on tizen

To use fvp
- flutter must expose texture api in c/c++, for example flutter for linux has fl_texture_gl.h for developers, but flutter-tizen does not
- All the dependencies have to be compatible with tizen. Compiling is the easiest work, but still takes a lot of time to writing patches and compiling
- tizen lacks of some standard linux libraries, for example video decoder. instead it provides a new api set, which means we have to write tizen specific code. 

developing for tizen support is a huge work, it's not possible to support tizen easily unless you can provide resources for us.
